### DISCUSSION TITLE
Implementing Option 2 from "Search With Permissions" page

### DISCUSSION THREAD

**Discussion by a0js** [openfga/community#277] [Q&A]
I need to implement [option 2](https://openfga.dev/docs/interacting/search-with-permissions#option-2-build-a-local-index-from-changes-endpoint-search-then-check) for our use case where we are trying to return lists of objects that a user has permission to access/do some action on. We have a graphql service that is retrieving data from OpenFGA to determine which search results should be returned based on authorization. I have a couple of questions questions:

- Is the "Build A Local Index From Changes Endpoint" something we would do in the graphql service or in a layer between graphql and openfga? 
- By "Local Index", is that suggesting an in-memory index or something like a redis cache? Would a redis cache allow for sharing the index across load balanced instances of our services?
- Do we hit `/changes` only on the startup of the service to build the index, or do we do it on each request to keep the index up to date? We just store the continuation token and use that for each call?
 
I'm just looking for a little more guidance and ideas from what others have done to implement this suggestion as it is arguably the most vague of the 3 options.

Edit: Also, doesn't it feel odd to create a local index of the tuples in the database in the first place? Is there an extension we can bolt onto openfga that supports "intesectioning" so we don't have to spin up our own?

**Comment by aaguiarz** [comment]
Hi @a0js!

We have [this](https://github.com/openfga/roadmap/issues/51) in our roadmap that will try to address that use case. Let me know if you think it can help for your scenario

Also, we want to learn about how we can help GraphQL customer to better use OpenFGA. If you are willing to spend 30 min with us to share how you are using it, please reach out to andres.aguiar at openfga.dev :)

Thanks!
